refinery/tool_manager/utils.py

```diff
@@ -468,14 +468,18 @@ def parse_file_relationship_nesting(nested_structure, nesting_dict=None,
         raise RuntimeError(
             "LIST/PAIR structure is not balanced {}".format(nesting_contents)
         )
+    # TODO: If it's a unicode instead of str, it breaks. Too fragile?
```

Comment:
I'm fine with adding `unicode` to this check

Reply:
Filed #2108: Hopefully not hard, but I could imagine it blowing up.